### PR TITLE
first draft of 'first responder' guidelines doc

### PR DESCRIPTION
**Pull request by jmartin-sul**
h/t to @ndushay for writing the initial google doc that got edited into this markdown doc.

i could still do another full proof-read of this doc.

in the meantime, feedback welcome, of course!

### CODE REVIEW

**Comment by jcoyne** [comment]
Thank you for putting together such a comprehensive guide. 👏

File: first_responder.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+The infrastructure team has 8 developers, so you should be taking a shift every 8 or so weeks.
+* schedule:  https://docs.google.com/spreadsheets/u/1/d/13TJR93Yc9_eF5B7w4XDx6ggG_wb3aLkgCHjpLwmHPBA/
+
+### Proactively Check for Production Problems
+
+See "Proactively Check for Production Problems" section below for specifics. 
+
+### Triage Production Problems
+
+If a user reports a problem, or if one is surfaced from monitoring, the first responder is meant to timebox an investigation of the problem (_TBD: 30 min?_).  See "Triage Production Problems" section for more specifics.
```
Comment (jmartin-sul) [inline first_responder.md]: the rule of thumb for the amount of investigation time is something i'm guessing we'll figure out as we try this, but FYI, open question...

File: first_responder.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+* Slack channels relevant to applications in the portfolio, e.g. `#pre-assembly`
+* resque dashboards?  e.g. for preservation catalog?
+* cron daemon emails
+* Github security alerts
+* If you hear about possible security issues through other avenues (e.g. `#iso-public` on Slack, your consumption of the news in general), and you feel that the issues may be relevant and uncaptured, file an issue and call attention to it as needed.
+
+
+## Triaging Production Problems
+
+If a user reports a problem, or if one is surfaced from monitoring, the first responder is meant to timebox an investigation of the problem (_TBD: no more than 30 min?_).  It's fine to ask teammates with relevant expertise for help, but also think about what documentation is needed so the next person will need less help. It is _not_ the job of the first responder to fix the issue on the spot (though if the fix is trivial, it's fine to do so).
```
Comment (jmartin-sul) [inline first_responder.md]: same timebox question

File: first_responder.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+It *may* be that someone in a time zone further east than Palo Alto has already done this, but it is the first responder's responsibility to ensure this gets done.
+
+#### Ensure Monday dependency updates are completed
+
+It is a team task to complete these updates, but the first responder needs to make sure that all codebases needing updates have updates merged and deployed.  Note that some projects need to have PRs created by hand ("Manually clone and update repos that failed bundle update" on line 6 of instructions). It may be helpful to post in channel how many updates each developer should do, given who is working that day and how many PRs there are.
+
+### Verify / Notify coverage for following week
+
+1. Verify first responder for following week is still able to cover it.  Check this on Monday (though it's a good thing to keep in mind throughout the week, in case e.g. unexpected illnesses come up).
+  1. It's the scheduled responder's responsibility to find a swap, not current first responder. _TBD: We'll figure something out if said responder is unavailable due to emergency?_
```
Comment (jmartin-sul) [inline first_responder.md]: hopefully this won't come up often, and won't be too difficult to deal with if it does, but this isn't something we have a solid procedure for right now.

File: first_responder.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+Other places to monitor:
+* Nagios alerts
+* feedback email lists, e.g. argo-feedback, sinopia-feedback, etc
+* the `#dlss-infrastructure` channel
+* Slack channels relevant to applications in the portfolio, e.g. `#pre-assembly`
+* resque dashboards?  e.g. for preservation catalog?
+* cron daemon emails
+* Github security alerts
+* If you hear about possible security issues through other avenues (e.g. `#iso-public` on Slack, your consumption of the news in general), and you feel that the issues may be relevant and uncaptured, file an issue and call attention to it as needed.
+
```
Comment (jmartin-sul) [inline first_responder.md]: other tips for places to watch are welcome (@ndushay seemed esp interested in getting @jcoyne and/or @justinlittman to add anything they know of that might be missing)

File: first_responder.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+* Respond to production issues in a timely fashion during business hours.
+* Single process for handling questions outside the team's current work cycle.
+* Share the responsibility for production outages across the team in a planned way.
+* Every team member rotates through this responsibility.
+* Encourage cross-training, since a first-responder will likely have to investigate applications with which they are unfamiliar.
+* Shore up missing and outdated documentation of production code and processes for everyone (ops, devops, PSM, stakeholders, devs, etc), informed by actual attempts to find and use said documentation to investigate production issues.
+
+### "First responder" != "on call"
+
+* The idiosyncratic name of the role is intentional.  It is not the duty of the "first responder" to be on-call after hours.  The first responder rotation is an effort to watch and triage production issues in an intentional and organized fashion, since no engineer was officially assigned this responsibility in the past, and such minding of things was haphazard.
```
Comment (justinlittman) [inline first_responder.md]: Should indicate that this is during a first responders normal working hours, i.e., a first responder in a different time zone may monitor during Stanford business hours, but does not have an extraordinary requirement while on duty.

Comment (jmartin-sul) [reply first_responder.md]: fixed, thanks for the suggestion!

**Pull request merged**